stablehlo/tests/CheckOps.td

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2023 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_DIALECT_CHECK_OPS
+#define STABLEHLO_DIALECT_CHECK_OPS
+
+include "stablehlo/dialect/Base.td"
+
+//===----------------------------------------------------------------------===//
+// CHECK dialect.
+//===----------------------------------------------------------------------===//
+
+def CHECK_Dialect : Dialect {
+  let name = "check";
+  let cppNamespace = "::mlir::stablehlo::check";
+
+  let description = [{
+    Opset to implement test assertions for StableHLO interpreter.
+  }];
+
+  let useDefaultAttributePrinterParser = 0;
+  let useDefaultTypePrinterParser = 0;
+  let useFoldAPI = kEmitFoldAdaptorFolder;
+}
+
+//===----------------------------------------------------------------------===//
+// CHECK op definitions.
+//===----------------------------------------------------------------------===//
+
+def CHECK_EqOp :
+    Op<CHECK_Dialect, "eq", [AllTypesMatch<["lhs", "value"]>]> {
+  let summary = [{Checks that the tensor operand is equal to some constant}];
+  let description =  [{
+    Verifies that the tensor operand is exactly equal to a constant attribute.
+    For floating-point and complex types, uses bitwise comparison for equality.
+
+    Issues a non-fatal failure if the verification fails.
```

Comment:
Can we get some negative test cases for both the new ops? With only positive test cases I don't think we test that the "approximately equal" works, and to what tolerance - in the past I had implemented a float comparison function that unintentionally always returned true, which went uncaught for awhile because of inadequate negative testing.

Reply:
Also I only skimmed the tests and searched for what the error would be, if there are negative tests that I missed let me know and feel free to resolve the comment

Reply:
The following example demonstrates the tolerance in action. I have updated the reference.md with this information. Please have a look.

```mlir
func.func @check_tolerance() {
  %0 = stablehlo.constant dense<0.20000000000000001110> : tensor<f32>
  check.almost_eq %0, dense<0.199999988> : tensor<f32> // aseert true
  check.eq %0, dense<0.199999988> : tensor<f32> // asset false
  func.return
}
```

>> negative test

That is a good point. 
In the current implementation we giving  fatal error when the mismatch happens. With that I am not sure if I am unsure if I can add a negative test in the test suite.